test/compilable/testcolor.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+# We never use ANSI codes on Windows, so, we can't test them.
+if [[ $OS = *"win"* ]]; then exit 0; fi
+
+compare()
+{
+    local actual="$1"
+    local expected="$2"
+
+    if [ "$actual" != "$expected" ] ; then
+        printf 'Expected: %q\n' "$expected"
+        printf 'Actual:   %q\n' "$actual"
+        hexdump -C <<<"$expected"
+        hexdump -C <<<"$actual"
+        exit 1
+    fi
+}
+
+normalize() { tr -d "\n\r" ; }
+
+check()
+{
+    local actual expected
+    actual=$(echo "$2" | ("$DMD" -c -o- "$1" - 2>&1 || true) | normalize)
+    compare "$actual" "$3"
+}
+
+expectedWithoutColor=__stdin.d\(2\):\ Error:\ no\ identifier\ for\ declarator\ \`test\`
+expectedWithColor=$'\033'\[1m__stdin.d\(2\):\ $'\033'\[1\;31mError:\ $'\033'\[mno\ identifier\ for\ declarator\ $'\033'\[0\;36m$'\033'\[m$'\033'\[1mtest$'\033'\[0\;36m$'\033'\[m
+
+check -c "test" "$expectedWithoutColor"
+check -color=auto "test" "$expectedWithoutColor"
+check -color=on "test" "$expectedWithColor"
+check -color=off "test" "$expectedWithoutColor"
+
+if [[ "$(script --version)" == script\ from\ util-linux\ * ]]
+then
+    actual="$(! SHELL="$(command -v bash)" script -q --return -c "echo test | $DMD -c -o- -" /dev/null | normalize)"
```

Comment:
This doesn't seem to work on SemaphoreCi:

```
+++ script --version
++ [[ script from util-linux 2.20.1 == script\ from\ util-linux\ * ]]
+++ normalize
+++ tr -d '\n\r'
++++ command -v bash
+++ SHELL=/bin/bash
+++ script -q --return -c 'echo test | ../generated/linux/release/64/dmd -c -o- -' /dev/null
++ actual='__stdin.d(2): Error: no identifier for declarator test'
+ finish 1
```

We could try to do what I did in https://github.com/dlang/dmd/pull/7822/files?

Reply:
> This doesn't seem to work on SemaphoreCi:

I think that's due to a difference in bash version.